### PR TITLE
Read instructions from richtext entities on assignment/quiz

### PR DESCRIPTION
**Pull request by CodeBaboon**
[US90636 - Description for quizzes](https://rally1.rallydev.com/#/89963236876d/detail/userstory/156522784044)

Requires: 
https://git.dev.d2l/projects/CORE/repos/le/pull-requests/8375/overview
https://git.dev.d2l/projects/CORE/repos/le/pull-requests/8366/overview

Ultimately adds functionality to display quiz description when provided by the API. Due to API changes to the assignment instructions performed in le/8375 there is also modification to the handling of those as well.

### CODE REVIEW

File: behaviors/d2l-upcoming-assessments-behavior.html [hunk 1]
```diff
@@ -78,6 +78,16 @@
 			return this._fetchEntity(activityLink, getToken, userUrl);
 		},
 
+		_getAssignmentInstructions: function(activity) {
+			var instructions = activity.getSubEntityByRel('https://assignments.api.brightspace.com/rels/instructions');
```
Comment (ryantmer) [inline behaviors/d2l-upcoming-assessments-behavior.html]: I assume this is just waiting for the `d2l-hm-constants` PR, ja?

File: test/behaviors/d2l-upcoming-assessments-behavior.js [hunk 2]
```diff
@@ -211,17 +256,29 @@ describe('d2l upcoming assessments behavior', function() {
 				});
 		});
 
-		it('should fall back to the activity instructions property if instructionsText is null', function() {
-			var activityWithInstructions = JSON.parse(JSON.stringify(activity));
-			activityWithInstructions.properties.instructionsText = null;
-			activityWithInstructions.properties.instructions = 'some other text';
+		it('should set the instructions to the value returned from _getAssignmentInstructions if the activity is an assignment', function() {
+			component._getAssignmentInstructions = sandbox.stub().returns('bonita bonita bonita');
+			component._getQuizDescription = sandbox.stub().returns('time for new flava in ya ear');
```
Comment (ryantmer) [inline test/behaviors/d2l-upcoming-assessments-behavior.js]: I quite enjoy all these subtle references to 90s rap/hip-hop that you include in your changes

File: behaviors/d2l-upcoming-assessments-behavior.html (hunk 1 above)
Comment (CodeBaboon) [reply behaviors/d2l-upcoming-assessments-behavior.html]: Ja.

File: behaviors/d2l-upcoming-assessments-behavior.html [hunk 3]
```diff
@@ -109,11 +119,17 @@
 						var organization = response[1];
 
 						var type = self._getActivityType.call(self, activity);
+						var instructions;
+						if (type === 'quiz') {
+							instructions = self._getQuizDescription.call(self, activity);
```
Comment (ChrisKraljevicD2L) [inline behaviors/d2l-upcoming-assessments-behavior.html]: Nit: maybe `instructions` isn't the best name for this property/variable if we're going to be storing a "description" if the activity is a quiz.

**Comment by ChrisKraljevicD2L** [comment]
Code Review Checklist
- [ ] Verify that unit tests and spec tests have been written, if appropriate
- [ ] Verify that CI checks are passing
- [ ] Verify that the code is behind a feature flag
- [ ] Verify that appropriate documentation has been done (readme, inline, docs.dev, etc.) and makes sense
- [ ] Verify to the best of your ability that no security/privacy flaws are present
- [ ] Verify that accessibility standards are held to
- [ ] Verify that no lang terms are hard coded
- [ ] Verify that lang terms have been submitted for translation
- [ ] Other review for coding style, functionality, and best practices

Testing Checklist
Sanity Checks
- [ ] Verify that unit tests and spec tests have been written, if appropriate
- [ ] Verify that unit tests, spec tests, and linting are all passing
- [ ] Verify that no code review comments are outstanding
- [ ] Verify that the code is behind a feature flag

Acceptance/Functionality
- [ ] Check that the acceptance criteria of the story has been fulfilled
- [ ] Check that related functionality has not been severely affected (smoke test)
- [ ] (UI) Check that the functionality is the same across browsers and devices

Accessibility (UI)
- [ ] Verify that workflows can be completed via screen reader/keyboard
- [ ] Verify that workflows can be completed when the browser is zoomed to 200%
- [ ] Verify that there are no obvious colour contrast/combination problems

Internationalization (UI)
- [ ] Verify that lang terms are not hard coded
- [ ] Verify that workflows can be completed when RTL

**Comment by CodeBaboon** [comment]
Too bad Travis seems to be stuck. Gonna try restarting the builds...

**Comment by CodeBaboon** [comment]
GH not reflecting it, but the [build for db4cebe](https://travis-ci.com/Brightspace/d2l-upcoming-assessments-ui/builds/56004353) passed. Going to go ahead and merge this.

**Pull request merged**